x-pack/plugins/security_solution/common/api/endpoint/agent/get_agent_status_route.ts

```diff
@@ -0,0 +1,41 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { schema, type TypeOf } from '@kbn/config-schema';
+import { RESPONSE_ACTION_AGENT_TYPE } from '../../../endpoint/service/response_actions/constants';
+
+const AgentStatusAgentIdSchema = schema.string({
+  minLength: 1,
+  validate: (id) => {
+    if (id.trim() === '') {
+      return 'actionIds can not be empty strings';
+    }
+  },
+});
+
+export const EndpointAgentStatusRequestSchema = {
+  query: schema.object({
+    agentIds: schema.oneOf([
+      schema.arrayOf(AgentStatusAgentIdSchema, { minSize: 1, maxSize: 50 }),
+      AgentStatusAgentIdSchema,
+    ]),
+
+    agentType: schema.maybe(
```

Comment:
This would limit the agent details/status search to one agent type. We should make this so it accepts multiple agent types.

Reply:
Adding support for multiple agent types would likely need to convert this API to a `POST` with body and then have an array of input with `{ agentType: string; agentId: string }` - making the API much more complex and perhaps harder to work with. Lets hold off on that until the need arises. My thinking is that the caller would issue two api calls to get the status, but perhaps your suggestion would work best when we start to introduce support for many external EDR systems. I think the real need will come when we start to show a list of Agents/Hosts from multiple EDR system in one view (similar to our Endpoint list) where we need to show the status.